### PR TITLE
SPR-147 improve commenting and documentation

### PR DESCRIPTION
**Pull request by dbarkowsky**
# Description

This PR includes the following proposed change(s):

- Adding jsdoc comments everywhere.
- Updating Swagger yaml to reflect recent schema changes.
- Realized my TESTING env in Constants somehow ended up with backwards logic. Fixed.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change
- [ ] Dependencies added/removed

## Aspect(s) of Project Affected
- [ ] Frontend
- [x] API
- [ ] Database
- [ ] Workflows
- [x] Documentation

## Best Practices Checklist
- [x] I have performed a self-review of my own code
- [x] My code follows the Airbnb React Style Guidelines
- [x] Documentation has been updated to reflect my changes
- [x] New and existing tests pass locally with my changes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Tested Swagger updates through local Swagger page.

### CODE REVIEW

File: api/controllers/chefs-api-controller.ts [hunk 1]
```diff
@@ -6,21 +6,36 @@ import RequestStates from '../constants/RequestStates';
...
 const removeBlankKeys = (obj: object) => {
   Object.keys(obj).forEach((key: keyof object) => {
     if (obj[key] === '') delete obj[key];
   });
   return obj;
 }
 
+/**
+ * @description Takes requests from the CHEFS service and submits them to the database.
+ * @param req The incoming CHEFS request
```
Comment (BradyMitch) [inline api/controllers/chefs-api-controller.ts:35]:
```suggestion
 * @param {ChefsRequest} req The incoming CHEFS request
```

File: api/controllers/chefs-api-controller.ts [hunk 2]
```diff
@@ -6,21 +6,36 @@ import RequestStates from '../constants/RequestStates';
...
+ * @param res The outgoing response
```
Comment (BradyMitch) [inline api/controllers/chefs-api-controller.ts:36]:
```suggestion
 * @param {Response} res The outgoing response
```

File: app/src/App.tsx [hunk 3]
```diff
@@ -11,6 +11,10 @@ import { KeycloakWrapper } from './keycloak';
 import { useAuthService } from './keycloak';
 import Login from './pages/Login';
 
+/**
+ *
```
Comment (BradyMitch) [inline app/src/App.tsx:15]:
```suggestion
```

**Review by BradyMitch — APPROVED** [review]
I suggest adding the types to your @param comments

**Review by Bottle7 — APPROVED** [review]

**Comment by dbarkowsky** [comment]
> I suggest adding the types to your @param comments

Added types to every param I could find. Will need re-review.

File: app/src/App.tsx (hunk 3 above)
Comment (dbarkowsky) [reply app/src/App.tsx]: Addressed in param commit

File: api/controllers/chefs-api-controller.ts (hunk 2 above)
Comment (dbarkowsky) [reply api/controllers/chefs-api-controller.ts]: Added

File: api/controllers/chefs-api-controller.ts (hunk 1 above)
Comment (dbarkowsky) [reply api/controllers/chefs-api-controller.ts]: Added

**Pull request merged**